app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -330,6 +330,8 @@ def process_vm_hardware(vm)
     end
 
     def process_snapshots(p)
+      return unless p[:VMCheckpoints]
```

Comment:
This will return `nil` from this method.  Will the caller know how to handle `nil` vs `[]`?